routes/courseRoutes.js

```diff
@@ -307,21 +305,17 @@ router.delete("/:id"/*, requireLogin*/, async (req, res) => {
     lectureIds.map(async (lecture_id) => {
 
       // Delete the lecture
-      await LectureModel.findByIdAndDelete( lecture_id, (err) => {
+      await LectureModel.findByIdAndDelete(lecture_id).catch((err) => res.status(422).send(err));
```

Comment:
Is 422 the correct response status?

The API guidelines specify this:
_"DO return a 204-No Content without a resource/body for a DELETE operation (even if the URL identifies a resource that does not exist; do not return 404-Not Found)"_

I think you should double check this, as I am not too sure.